File: 1365.how-many-numbers-are-smaller-than-the-current-number.java
/*
 * @lc app=leetcode id=1365 lang=java
 *
 * [1365] How Many Numbers Are Smaller Than the Current Number
 */

// @lc code=start
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] times = new int[101];
        for (int ix = 0; ix != nums.length; ++ix) {
            times[nums[ix]] += 1;
        }
        for (int ix = 1; ix != times.length; ++ix) {
            times[ix] += times[ix - 1];
        }
        int[] ret = new int[nums.length];
        for (int ix = 0; ix != ret.length; ++ix) {
            ret[ix] = (nums[ix] == 0) ? 0 : times[nums[ix] - 1];
        }
        return ret;
    }
}
// @lc code=end

File: 1365.how-many-numbers-are-smaller-than-the-current-number.cpp
/*
 * @lc app=leetcode id=1365 lang=cpp
 *
 * [1365] How Many Numbers Are Smaller Than the Current Number
 */

// @lc code=start

#include <vector>

using namespace std;

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> t(101, 0);
        for (auto el : nums) t[el] += 1;
        int acc = 0;
        for (int ix = 0; ix != 101; ++ix) {
            int tmp = t[ix];
            t[ix] = acc;
            acc += tmp;
        }
        vector<int> ret(nums.size(), 0);
        for (int ix = 0; ix != ret.size(); ++ix) {
            ret[ix] = t[nums[ix]];
        }
        return ret;
    }
};
// @lc code=end
